pkg/dns/azure/client/auth.go

```diff
@@ -73,15 +75,33 @@ func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
 	// Managed Identity Override for ARO HCP
 	managedIdentityClientID := os.Getenv("ARO_HCP_MI_CLIENT_ID")
 	if managedIdentityClientID != "" {
-		options := azidentity.ManagedIdentityCredentialOptions{
+		options := &azidentity.ClientCertificateCredentialOptions{
 			ClientOptions: azcore.ClientOptions{
 				Cloud: cloudConfig,
 			},
-			ID: azidentity.ClientID(managedIdentityClientID),
+			SendCertificateChain: true,
 		}
 
-		var err error
-		cred, err = azidentity.NewManagedIdentityCredential(&options)
+		tenantID := os.Getenv("ARO_HCP_TENANT_ID")
```

Comment:
What is the relationship between the tenant id (`config.TenantID` / `AZURE_TENANT_ID`) and the HCP tenant id (`ARO_HCP_TENANT_ID`)?  `getAuthorizerForResource` returns an error if the tenant id isn't set, but it looks like the tenant id isn't ultimately used if the HCP tenant id is used.  Does HCP just set a dummy value for tenant id?  Will this be simplified in the future?  For now, can you add a comment explaining that the tenant id is ignored if the HCP tenant id is used (or clarify if I misunderstood the logic)?

Similarly, it looks like the client id is ignored when using the HCP client id.

Reply:
Yeah we will just be passing a dummy client ID/tenant ID through the required secret - https://github.com/openshift/hypershift/pull/4587/commits/37f837af562216ce71857be86dd8e4ad2099cc6a#diff-c888020c5dc46c458d818b931db97131d6e35b90661fe1030a39ebeac8859b19R1443 

`AZURE_TENANT_ID` and `AZURE_CLIENT_ID` ultimately is not used in ARO HCP cases for the operator. We are still reconciling the needed secret since its required in ingress. The ingress controller fails if the secret is not provided. The ingress controller runs on the control plane side in HCP. In managed azure, we are overriding the Azure credentials authentication method to always use client certificate authentication. The secret is just created so that the ingress controller does not fail; the data in the secret is never used due to the aforementioned override to use client certificate authentication.